pydantic_ai_slim/pydantic_ai/capabilities/combined.py

```
@@ -74,6 +75,16 @@ def has_wrap_node_run(self) -> bool:
     def has_wrap_run_event_stream(self) -> bool:
         return any(c.has_wrap_run_event_stream for c in self.capabilities)
 
+    @property
+    def has_resolve_model_id(self) -> bool:
+        return any(c.has_resolve_model_id for c in self.capabilities)
+
+    def for_agent(self, agent: AbstractAgent[AgentDepsT, Any]) -> CombinedCapability[AgentDepsT]:
+        new_caps = [c.for_agent(agent) for c in self.capabilities]
+        if all(new is old for new, old in zip(new_caps, self.capabilities)):  # pragma: no branch
```

Ruff (0.15.20)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)